app/controllers/transactions_controller.rb

```diff
@@ -155,7 +155,23 @@ def show
     }
   end
 
-  def op_status
+  def transaction_op_status
+    process_token = params[:process_token]
+
+    resp = Maybe(process_token)
+      .map { |ptok| transaction_process_tokens.get_status(ptok) }
+      .select(&:success)
+      .data
+      .or_else(nil)
+
+    if resp
+      render :json => resp
+    else
+      redirect_to error_not_found_path
```

Comment:
I know this is copy&paste code but it really does not make any sense to redirect to a 404 since this is a JSON endpoint and nobody will even see the error page.